### PR TITLE
[MM-22133] Allow exporting and importing archived channels

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
Allows exporting and importing archived public and private channels from the mattermost cli bulk import/export and mmctl. **does not support Slack for this first version**

- `mmctl export create` now has a `--include-archived-channels` that will make the export include archived channels
- `mattermost export bulk` also has a `--with-archived-channels` that does the same.
- both import (from mmctl or mattermost import bulk` will insert create channel.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22133

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/i18n/en.json [hunk 1]
```diff
@@ -5235,6 +5235,10 @@
     "id": "app.import.get_users_by_username.some_users_not_found.error",
     "translation": "Some users not found"
   },
+  {
+    "id": "app.import.import_channel.deleting.app_error",
+    "translation": "Unable to archive importer channels."
```
Comment (pvev) [inline server/i18n/en.json:5240]: does this have a typo?

File: server/channels/store/sqlstore/channel_store.go [hunk 2]
```diff
@@ -1517,7 +1525,9 @@ func (s SqlChannelStore) GetByNames(teamId string, names []string, allowFromCach
 			visited[name] = struct{}{}
 			var cacheItem *model.Channel
 			if err := channelByNameCache.Get(teamId+name, &cacheItem); err == nil {
-				channels = append(channels, cacheItem)
+				if includeArchivedChannels || cacheItem.DeleteAt == 0 {
```
Comment (pvev) [inline server/channels/store/sqlstore/channel_store.go:1528]: shouldn't this logic be an && instead of an || ?

**Review by pvev — APPROVED** [review]
LGTM! thanks @JulienTant! :shipit:

File: server/channels/store/sqlstore/channel_store.go (hunk 2 above)
Comment (JulienTant) [reply server/channels/store/sqlstore/channel_store.go:1528]: In every case, if DeleteAt is 0 we want to add the channel to the list of found. And if `includeArchivedChannels` is `true`, we can ignore the value of `DeleteAt`. So `||` wins! lol

**Comment by JulienTant** [comment]
reminder about this @DHaussermann - let me know if I can help in any way!

**Comment by DHaussermann** [comment]
@JulienTant sorry for the delay. I've never used the cannel import/export. Let me try and familiarize myself with it a bit. I will reach out if I have questions.

**Comment by DHaussermann** [comment]
Hey @JulienTant this task is clear to me but I wonder if you can assists with setup. 

I don't know how to go about logging in to create the backup...
image

I created a new Docket container from the image on this commit. I then opened a shell into the container to try the commands so that I can use the MMCTL tool. 
But because it's a docker with no mounted volume, I cannot add or manipulate config files anywhere.   

Is there a way to login from the command line where I don't need to change anything on the file system?
Is there a simpler way to just test this? Perhaps I can get the appropriate MMCTL tool somewhere else?

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Confirmed you can pass in `--include-archived-channels` to include archive channels
- Importing the data back in does respect that the channels are archived and imports them that way.
- Also test bulk export and passed in the `--with-archived-channels` and channels are marked as archived in the `.json`
 - Bulk import of data correctly creates archived channels during the import process
 
 LGTM!

**Pull request merged**